### PR TITLE
add launcher type for specifying cluster operations locally or via Den

### PR DESCRIPTION
**Pull request by jlewitt1**

### CODE REVIEW

**Comment by jlewitt1** [comment]
* **#1336** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1336?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jlewitt1 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/hardware/cluster_factory.py [hunk 1]
```diff
@@ -50,6 +51,9 @@ def cluster(
             API server. ``ssh`` will use start with server via an SSH tunnel. ``tls`` will start the server
             with HTTPS on port 443 using TLS certs without an SSH tunnel. ``none`` will start the server with HTTP
             without an SSH tunnel. (Default: ``None``).
+        launcher_type (LauncherType or str, optional): Method for launching the cluster. If set to `local`, will launch
+            locally via Sky. If set to `runhouse`, launching will be handled by Runhouse. Currently only relevant for
```
Comment (carolineechen) [inline runhouse/resources/hardware/cluster_factory.py:55]: this should be `den` not `runhouse` right?

File: runhouse/resources/hardware/cluster.py [hunk 2]
```diff
@@ -710,9 +710,10 @@ def _sync_runhouse_to_cluster(
             # rh_package = 'runhouse_nightly-0.0.1.dev20221202-py3-none-any.whl'
             # rh_download_cmd = f'curl https://runhouse-package.s3.amazonaws.com/{rh_package} --output {rh_package}'
             if not _install_url:
-                import runhouse
+                # import runhouse
 
-                _install_url = f"runhouse=={runhouse.__version__}"
+                # _install_url = f"runhouse=={runhouse.__version__}"
+                _install_url = "git+https://github.com/run-house/runhouse.git@replace-sky-with-den-apis#egg=runhouse[all]"
```
Comment (jlewitt1) [inline runhouse/resources/hardware/cluster.py:716]: @mkandler are we still using this branch for anything? can we revert this?

**Review by mkandler — APPROVED** [review]
LGTM

**Pull request merged**